subsys/bluetooth/controller/Kconfig.ll_sw_split

```diff
@@ -528,14 +528,32 @@ config BT_TICKER_COMPATIBILITY_MODE
 
 config BT_TICKER_EXT
 	bool "Ticker extensions"
-	depends on !BT_TICKER_COMPATIBILITY_MODE
+	depends on !BT_TICKER_COMPATIBILITY_MODE && !BT_TICKER_SLOT_AGNOSTIC
```

Comment:
Good idea with the rename.
> 2. Is slot_window still supported?

No, this is something that needs to be reworked in the ull_adv (and ull_scan if needed). Since there is no notion of slot reservation and collision, the slot_window cannot be used to reschedule within a window. For ADV, it's reasonably simple to implement the same "windowing" mechanics in LL, if it does not get to run. I have a solution for that.

Reply:
What about a replacement for `ticker_next_slot_get`, this is something currently used to fill Connection Parameter Request offsets, schedule non-overlapping central  connections, Auxiliary Pointer offset, Sync Info offset, BIGInfo offset etc... Do you plan to supply/share the time reservation/usage by each role in the new framework?

Reply:
> What about a replacement for `ticker_next_slot_get`

Yes, this is a very good question. I am aware of the `ticker_next_slot_get` functionality. Even though I must say I don't like it very much, I agree it is needed for the purposes you mention.
So how to handle this without the slot reservation? Well, the function only uses the slot for finding ticker nodes that care about "air-time" - it doesn't use the slot reservation as such. It should be a small change to have it search for maybe a specific ticker node id or range. The result is anyway checked against ticker id and disqualified in some cases, causing a new iteration of searching. The best solution is of course to have the searching find the right ticker in the first attempt, and that can probably be implemented by passing a "search criterion".
The need for locating optimal ticker timing "locations" is growing. Vendors will be needing to request the best possible location for placing central connections, to allow space for large audio packets to two or more peripherals.
But tell me, did I miss some important fact here?

Reply:
>  Vendors will be needing to request the best possible location for placing central connections, to allow space for large audio packets to two or more peripherals.

`ticker_next_slot_get` is an enumeration interface, in a primitive form which is used currently to place (for now central, but shall be used to group auxiliary and periodic PDUs too).

The ticker concept is to keep time reservations in a central location (other concept is, distributed in users of ticker, in advertisers/scanner/central/peripheral etc., needing a framework across users adding to a variable execution time dependencies based on how a role is decided to be implemented in the framework), such that its scalable with respect to users. These time reservations shall be the bare minimal to guarantee the minimal CPU/Radio and other resource availability.

Same applies to priorities too, keep it centralized (as in real hardware SoC's, interrupt controllers). `score` concept seems inline towards fairness when `abort` or `too late` happens in LLL pipeline. I do not like distributed dynamic priorities, but weight or one time `force` as a binary decision between two tickers does solve the limited collision resolutions requirements like trying to avoid supervision timeouts in connections, periodic sync, BIG Sync and ISO connections. I see `score` in addition to `force` helping here. The decision to pre-empt shall be based on the combination of no. of `lazy_current` above the requested `lazy_periodic`, `score` which is the no. of consecutive `aborts` and `too lates`, and instantaneous applied `force`.  `score` can be reflected into `lazy_current`, if it is just to let ticker know that the timeout callback was skipped in the LLL pipeline.

> did I miss some important fact here?
Not sure. But, my concern is towards the duty cycle of collision resolutions, its complexity, scheduling overheads, and ability to share the time reservations to roles.

If its ok, lets do small incremental improvements/replacements. If its ok, we can start with introducing the concept of `score`, either using explicit `score` variable in ticker node, or translating it into equivalent `lazy_current` value.

Reply:
> These time reservations shall be the bare minimal to guarantee the minimal CPU/Radio and other resource availability.

My main goal of this exercise is to test the theory that we **don't need reservations at all**. The overall concept is that if an event gets to be programmed in the radio, there's a good chance it will finish - but maybe not. The priority takes over the role of reservation, such that a higher priority event may preempt a lower priority event, if they collide.
Actually I think it's only advertising, which has an (almost) precise reservation, and still it reserves space for scan req/rsp and connection ind, which more than often is not needed.

For the concept to work, we cannot even reserve the time for ramping up the radio, there cannot be reservations, because we must make sure that when the preemption timer elapses with a higher priority prepare, it always gets to start.

> The decision to pre-empt shall be based on the combination of no. of `lazy_current` above the requested `lazy_periodic`, `score` which is the no. of consecutive `aborts` and `too lates`, and instantaneous applied `force`. `score` can be reflected into `lazy_current`, if it is just to let ticker know that the timeout callback was skipped in the LLL pipeline.

Yes, I agree - except for `lazy_current`. The lazy parameter of the `lll_prepare_param` will only reflect periodic latency, since `lazy_current` is no longer incremented due to collisions. This also means that for instance the slave prepare is always called, since the decision to skip/queue the prepare is made _after_ the call to `lll_slave_prepare`. This also means that we may not need the `latency` variable that I added (but not yet used) - and that `score` becomes the old `lazy_current`, which again means we could just rename `score` to `lazy`, to stay in legacy terms(?) The idea is that the score/lazy value is added to the (periodic) lazy value from the ticker, and then passed to the prepare_cb in the `prepare_param`. This way, the link layer will "see" the same `lazy` value as for the old ticker.